src/cbmc_starter_kit/template-for-repository/proofs/lib/summarize.py

```diff
@@ -0,0 +1,113 @@
+# Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+# SPDX-License-Identifier: MIT-0
+
+import json
+import logging
+import sys
+
+
+def _get_max_length(column):
+    return max(len(word) for word in column)
+
+
+def _get_list_of_max_length_per_column(data):
+    list_of_max_length_per_column = []
+    num_rows = len(data)
+    num_columns = len(data[0])
+    for column_idx in range(num_columns):
+        column = [data[row_idx][column_idx] for row_idx in range(num_rows)]
+        max_length = _get_max_length(column)
+        max_length_with_extra_space = max_length + 1
+        list_of_max_length_per_column.append(max_length_with_extra_space)
+    return list_of_max_length_per_column
+
+
+def _get_header(list_of_max_length_per_column, header_names):
+    header = ""
+    for idx, word in enumerate(header_names):
+        max_length_of_word_in_col = list_of_max_length_per_column[idx]
+        space_formatted_word = (max_length_of_word_in_col - len(word)) * " "
+        header += "| " + word + space_formatted_word
+    header += "|\n"
+    return header
+
+
+def _get_line_separator_after_header(list_of_max_length_per_column):
+    line_sep = ""
```

Comment:
If you do as I suggested above with treating the headers the same as the data, it would make it easier to remove this function. Ideally if you 'render' the first line of the table as a string called `header`, then this would just be `print("-" * len(header)).

Reply:
> ...then this would just be `print("-" * len(header))`.

`print("-" * len(header))` will not work because the resulting table could be something like the following:

```
| Status  | Count |
-------------------
| Fail    | 1    |
| Success | 1    |
```

which could be rendered, if so chosen, as such:


| Status  | Count |
-------------------
| Fail    | 1    |
| Success | 1    |

Reply:
> ...then this would just be print("-" * len(header))`.

In the case of generating a table that follows the GitHub flavored Markdown spec and is left-aligned appropriately, the line that follows the header and separates the header from the "row entries" needs to be something like:

```
| --------- | ---- | ---------- |
```

where the number of dashes between the pipes is equal to the number of characters inside either the longest "row entry" or the given "header name" for that column.